### PR TITLE
Move pipelined example to snax-alu addition

### PR DESCRIPTION
**Pull request by JosseVanDelm**
This moves the tiled example to use the snax-alu accelerator.
For now, it just performs a 64-bit addition.
Done:
- [x] Move names to tiled_add instead of tiled_multiplication
- [x] Also test negative numbers
- [x] Rebase to main
- [x] Derive shape from memref instead of constant
- [x] Add offset support
- [x] Fix synchronization issue in tiled_pipelined example

### CODE REVIEW

File: kernels/tiled_mult/tiled_pipelined.preprocfinal.mlir [hunk 1]
```diff
@@ -89,38 +86,38 @@ func.func public @simple_mult(%A: memref<128xi32, "L3">,
...
+      %tiled_D_L1_1 = "memref.subview"(%D_L1, %last_index, %tile_size) <{operandSegmentSizes = array<i32: 1, 1, 1, 0>, static_offsets = array<i64:-9223372036854775808>, static_sizes = array<i64: -9223372036854775808>, static_strides = array<i64: 1>}> : (memref<128xi64, "L1">, index, index) -> memref<?xi64, strided<[1], offset: ?>, "L1">
+      "linalg.generic"(%tiled_A_L1_1, %tiled_B_L1_1, %tiled_D_L1_1) <{indexing_maps = [affine_map<(n) -> (n)>, affine_map<(n) -> (n)>, affine_map<(n) -> (n)>], iterator_types = [#linalg.iterator_type<parallel>], library_call = "snax_alu", operandSegmentSizes = array<i32: 2, 1>}> ({
+      ^bb0(%arg3: i64, %arg4: i64, %arg5: i64):
+        %10 = "arith.addi"(%arg3, %arg4) : (i64, i64) -> i64
+        "linalg.yield"(%10) : (i64) -> ()
+      }) : (memref<?xi64, strided<[1], offset: ?>, "L1">, memref<?xi64, strided<[1], offset: ?>, "L1">, memref<?xi64, strided<[1], offset: ?>, "L1">) -> ()
       // Wait for output to go from L1 to L3
       "snax.cluster_sync_op"() : () -> ()
+      "snax.cluster_sync_op"() : () -> ()
```
Comment (JosseVanDelm) [inline kernels/tiled_mult/tiled_pipelined.preprocfinal.mlir:120]: Moving down the `snax.cluster_sync_op` was the final issue here. It's likely to have been wrong previously as well.
Right now the final tile was executed and copied out in the same window, which could lead to sometimes right/sometimes wrong behaviour based on the speed of the accelerator and the DMA engine.
I found out because it worked when I inserted a printf in front of the DMA start...

**Review by AntonLydike — APPROVED** [review]
You are a machine my dude! Congrats! :mechanical_arm: :rocket:
